Plugins/NativeEngine/Source/NativeDataStream.h

```diff
@@ -0,0 +1,223 @@
+#pragma once
+
+#include <Babylon/JsRuntime.h>
+
+#include <napi/env.h>
+
+#include <gsl/gsl>
+
+namespace Babylon
+{
+    class NativeDataStream final : public Napi::ObjectWrap<NativeDataStream>
+    {
+        static constexpr auto JS_CLASS_NAME = "_NativeDataStream";
+        static constexpr auto JS_ENGINE_CONSTRUCTOR_NAME = "NativeDataStream";
+
+        static constexpr auto VALIDATION_ENABLED = false;
+
+        enum class ValidationType : uint32_t
+        {
+            Uint8,
+            Uint32,
+            Int32,
+            Float32,
+            Uint32Array,
+            Int32Array,
+            Float32Array,
+            NativeData,
+            Boolean,
+        };
+
+        template<ValidationType T, typename ReaderT>
+        static void Validate(ReaderT& reader)
```

Comment:
```suggestion
        static inline void Validate(ReaderT& reader)
```

Reply:
Is that necessary to make explicit? As a method defined in a header (let alone a static templated one), isn't it automatically `inline`?

Reply:
I don't know that all compilers will inline this. Regardless, it is a hint anyways. Not a big deal regardless.